pytorch_lightning/metrics/converters.py

```diff
@@ -279,6 +307,43 @@ def sync_ddp_if_available(
     return result
 
 
+def sync_horovod_if_available(
+    result: Union[torch.Tensor], group: Optional[Any] = None, reduce_op: Optional[Union[ReduceOp, str]] = None
+) -> torch.Tensor:
+    """
+    Function to reduce a tensor across all Horovod processes
+
+    Args:
+        result: the tensor to sync and reduce
+        group: the process group to gather results from. Defaults to all processes (world).
+            NOTE: Horovod does not support allreduce using communicators other than world
+            at this time.
+        reduce_op: the reduction operation. Defaults to sum.
+            Can also be a string of 'avg', 'mean' to calculate the mean during reduction.
+
+    Return:
+        reduced value
+    """
+
+    if HOROVOD_AVAILABLE and hvd.is_initialized():
+        if group is not None:
+            raise ValueError("Horovod does not support allreduce using a subcommunicator at "
+                             "this time. Unset `group`.")
+
+        if reduce_op is None or reduce_op == "sum":
+            reduce_op = hvd.Sum
+        elif isinstance(reduce_op, str) and reduce_op in ("avg", "mean"):
+            reduce_op = hvd.Average
+        else:
+            raise ValueError(f"unrecognized `reduce_op`: {reduce_op}")
+
+        # sync all processes before reduction
+        hvd.join()
+        result = hvd.allreduce(result, op=reduce_op)
+
+    return result
```

Comment:
the condition is already tested before
```suggestion
    if not (HOROVOD_AVAILABLE and hvd.is_initialized()):
        return result
    if group is not None:
        raise ValueError(
            "Horovod does not support allreduce using a subcommunicator at this time. Unset `group`."
        )

    if reduce_op is None or reduce_op == "sum":
        reduce_op = hvd.Sum
    elif isinstance(reduce_op, str) and reduce_op in ("avg", "mean"):
        reduce_op = hvd.Average
    else:
        raise ValueError(f"unrecognized `reduce_op`: {reduce_op}")

    # sync all processes before reduction
    hvd.join()
    return hvd.allreduce(result, op=reduce_op)
```